### PR TITLE
Prevent generating preview for zero sized texture

### LINKED ISSUE #93151: ERROR: The Image width specified (0 pixels) must be greater than 0 pixels.
### Tested versions

v4.3.dev6.mono.official [89850d553]

### System information

Godot v4.3.dev6.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 5GB (NVIDIA; 31.0.15.3623) - AMD Ryzen 9 3900X 12-Core Processor (24 Threads)

### Issue description

It report error when start project:
image
Note: it only report in the console, the Editor's output no error print.

### Steps to reproduce

just start the MRP.

### Minimal reproduction project (MRP)

[ddstest.zip](https://github.com/user-attachments/files/15834737/ddstest.zip)

Other steps:
1. create a project.
2. import a dds file.
3. create altastexture.
4. set atlastexture.texture with the dds
5. restart project with console

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes https://github.com/godotengine/godot/issues/93151

### CODE REVIEW

**Comment by akien-mga** [comment]
I'm not sure that's a good idea, given this:
```cpp
int AtlasTexture::get_width() const {
        if (region.size.width == 0) {
                if (atlas.is_valid()) {
                        return atlas->get_width();
                }
                return 1;
        } else {
                return region.size.width + margin.size.width;
        }
}

int AtlasTexture::get_height() const {
        if (region.size.height == 0) {
                if (atlas.is_valid()) {
                        return atlas->get_height();
                }
                return 1;
        } else {
                return region.size.height + margin.size.height;
        }
}
```

It seems like it's expected that a region of size 0 will fall back to the full size atlas.

**Comment by jsjtxietian** [comment]
I just had another idea of fixing this bug !

File: editor/plugins/editor_preview_plugins.cpp [hunk 1]
```diff
@@ -97,7 +97,9 @@ Ref<Texture2D> EditorTexturePreviewPlugin::generate(const Ref<Resource> &p_from,
 			return Ref<Texture2D>();
 		}
 
-		img = atlas->get_region(atex->get_region());
+		if (atex->get_region().get_area() != 0) {
+			img = atlas->get_region(atex->get_region());
+		}
```
[the remark this answers is not in the document]
Comment (jsjtxietian) [reply editor/plugins/editor_preview_plugins.cpp:102]: Thanks!

File: editor/plugins/editor_preview_plugins.cpp [hunk 2]
```diff
@@ -97,6 +97,10 @@ Ref<Texture2D> EditorTexturePreviewPlugin::generate(const Ref<Resource> &p_from,
 			return Ref<Texture2D>();
 		}
 
+		if (atex->get_region().get_area() == 0.0) {
```
Comment (AThousandShips) [inline editor/plugins/editor_preview_plugins.cpp:100]:
```suggestion
		if (!atex->get_region().has_area()) {
```

Comment (jsjtxietian) [reply editor/plugins/editor_preview_plugins.cpp:100]: Thanks!

**Review by Mickeon — APPROVED** [review]
This looks like the more sensible solution to me

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**